### DISCUSSION TITLE
Use inviteUserByEmail in application

### DISCUSSION THREAD

**Discussion by pichichi91** [supabase/supabase#1687] [Questions] [answered]
Hi

I am trying to build a SaaS with supabase (might turn that into a supabase starter later on, for people to quickly set up a starter with Supabase x NextJS x Stripe and deploy within a few minutes on Vercel) - now I've build an admin Dashboard where the admin could manage the users. Therefore I'd like to call inviteUserByEmail from my code, but I am always getting  `{ message: 'User not allowed', status: 401 } }`

Are there any examples on how to use them using the supabase client in supabase-js? 

Thanks in Advance!

**Comment by easy-street** [comment, marked as answer]
the `supabase.auth.api.inviteUserByEmail` method can be used with a supabase client created with the `service_role` key instead of the `anon` key:

**/lib/supabase.js**
```
import { createClient } from "@supabase/supabase-js";

export const supabaseAdmin = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY
);
```
**/pages/api/invite.js**
```
import { supabaseAdmin } from "../../lib/supabase";

export default async function invite(req, res) {
  ...
  const { data, error } = await supabaseAdmin.auth.api.inviteUserByEmail(
    req.body.email
  );
  ...
}
```
more info on the `service_role` key (including security considerations) in the [Auth Deep Dive](https://supabase.io/docs/learn/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase)

**Comment by devidevio** [comment]
Alternatively you can setup a PostgreSQL function, secure it with RLS and trigger the invite directly from the database, see more here:
* https://github.com/supabase/supabase/discussions/6055#discussioncomment-4742764
